### PR TITLE
[benchmark] [refactor] Move fill() and reduction() into Membound suite, calculate the geometric mean of the time results

### PR DESCRIPTION
**Pull request by yolo2themoon**
Related issue = #2601

### CODE REVIEW

**Comment by taichi-gardener** [comment]
|Membound.cuda|||||||||||geometric_mean|
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
|**data size**|4.0KB|16.0KB|64.0KB|256.0KB|1.0MB|4.0MB|16.0MB|64.0MB|256.0MB|1.0GB||
|**repeat**|1000|1000|1000|1000|1000|1000|100|100|100|100||
|kernel elapsed time(ms)|||||||||||
|fill.i32|0.0064|0.0064|0.0064|0.0064|0.0071|0.0197|0.0741|0.3776|1.4962|5.7979|0.0473|
|fill.i64|0.0048|0.0048|0.0064|0.0064|0.0068|0.0187|0.0625|0.2544|1.1105|4.3149|0.0395|
|fill.f32|0.0047|0.0048|0.0064|0.0065|0.0072|0.0194|0.0761|0.354|1.4972|5.8316|0.0445|
|fill.f64|0.0046|0.0047|0.0064|0.0064|0.0068|0.0187|0.0607|0.2484|1.1228|4.3031|0.0391|
|kernel elapsed time(ms)|||||||||||
|saxpy.i32|0.0047|0.0048|0.0064|0.0064|0.0071|0.0187|0.0626|0.2381|0.9955|4.7333|0.0392|
|saxpy.i64|0.0047|0.0048|0.0064|0.0064|0.0071|0.0176|0.0619|0.2281|0.9018|3.6659|0.0375|
|saxpy.f32|0.0047|0.0047|0.0063|0.0064|0.0071|0.0186|0.0626|0.2379|1.006|4.6756|0.0391|
|saxpy.f64|0.0048|0.0048|0.0064|0.0064|0.0071|0.0182|0.0618|0.228|0.8953|3.6375|0.0376|
|kernel elapsed time(ms)|||||||||||
|reduction.i32|0.0061|0.0061|0.0084|0.0086|0.0094|0.0216|0.066|0.2471|1.0224|4.2353|0.0457|
|reduction.i64|0.0696|0.0696|0.0885|0.1006|0.0765|0.0821|0.0964|0.2384|0.8909|3.4594|0.1692|
|reduction.f32|0.0102|0.0102|0.0146|0.0146|0.0146|0.0235|0.0681|0.2564|1.1079|4.8122|0.061|
|reduction.f64|0.2007|0.2007|0.2007|0.2007|0.2008|0.2047|0.2137|0.3072|1.0692|3.493|0.3322|

File: benchmarks/misc/run.py [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
+        for s in self.impls:
+            s.run()
+
+    def write_md(self):
         try:
-            with open(filename, 'r+') as f:
+            with open(self.filename, 'r+') as f:
                 f.truncate()  # clear the previous result
         except FileNotFoundError:
             pass
```
Comment (yuanming-hu) [inline benchmarks/misc/run.py:27]:
```suggestion
```

File: benchmarks/misc/run.py [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
-            for line in lines:
-                f.write(line + '\n')
-            f.close()
+        f = open(self.filename, 'a')
+        for arch in test_archs:
+            for impl in self.impls:
+                lines = impl.mdlines(arch)
+                for line in lines:
+                    f.write(line + '\n')
+        f.close()
```
Comment (yuanming-hu) [inline benchmarks/misc/run.py:34]:
```suggestion
        with open(self.filename, 'w') as f:
            for arch in test_archs:
                for impl in self.impls:
                    lines = impl.mdlines(arch)
                    for line in lines:
                        print(line, file=f)
```

**Review by yuanming-hu — APPROVED** [review]
LGTM in general! Let's fix some minor issues.

**Comment by taichi-gardener** [comment]
|Membound.cuda|||||||||||geometric_mean|
|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|:--:|
|**data size**|4.0KB|16.0KB|64.0KB|256.0KB|1.0MB|4.0MB|16.0MB|64.0MB|256.0MB|1.0GB||
|**repeat**|1000|1000|1000|1000|1000|1000|100|100|100|100||
|kernel elapsed time(ms)|||||||||||
|fill.i32|0.0064|0.0063|0.0064|0.0064|0.0071|0.0193|0.0741|0.3798|1.4996|5.8193|0.0473|
|fill.i64|0.0048|0.0047|0.0064|0.0064|0.0068|0.0188|0.0618|0.2524|1.1166|4.3151|0.0393|
|fill.f32|0.0047|0.0048|0.0064|0.0064|0.0073|0.0194|0.0761|0.3543|1.4933|5.8204|0.0445|
|fill.f64|0.0047|0.0047|0.0064|0.0064|0.0068|0.0187|0.0614|0.25|1.1188|4.3008|0.0392|
|kernel elapsed time(ms)|||||||||||
|saxpy.i32|0.0046|0.0048|0.0064|0.0065|0.0071|0.0186|0.0626|0.238|0.9982|4.7387|0.0392|
|saxpy.i64|0.0047|0.0047|0.0064|0.0064|0.0071|0.0186|0.0618|0.2284|0.9017|3.6801|0.0376|
|saxpy.f32|0.0048|0.0047|0.0063|0.0064|0.0071|0.0187|0.0624|0.238|1.0069|4.6887|0.0392|
|saxpy.f64|0.0047|0.0047|0.0063|0.0063|0.0071|0.0185|0.0619|0.2278|0.8951|3.6158|0.0373|
|kernel elapsed time(ms)|||||||||||
|reduction.i32|0.0062|0.0062|0.0084|0.0087|0.0105|0.0217|0.066|0.2481|1.0205|4.2465|0.0463|
|reduction.i64|0.0696|0.0696|0.1006|0.0759|0.0765|0.0799|0.1008|0.2574|0.8886|3.455|0.1681|
|reduction.f32|0.0102|0.0102|0.0146|0.0146|0.0146|0.0235|0.0666|0.2462|1.0268|4.3529|0.0595|
|reduction.f64|0.2007|0.2007|0.2007|0.2007|0.2007|0.2037|0.2134|0.299|0.9304|3.4591|0.3262|

**Pull request merged**